storage/kv/txn_pebble_test.go

```diff
@@ -0,0 +1,331 @@
+package kv
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"slices"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/ssvlabs/ssv/storage/basedb"
+)
+
+// Pebble DB txn requires its own set of tests because it has a different behaviour:
+// - it doesn't support concurrent transactions
+// - it doesn't support committing a transaction after it's been discarded
+
+func TestPebbleTxn_Commit(t *testing.T) {
+	db := newTestPebbleDB(t)
+	txn := db.Begin()
+
+	err := txn.Commit()
+	require.NoError(t, err)
+}
+
+func TestPebbleTxn_Discard(t *testing.T) {
+	db := newTestPebbleDB(t)
+	txn := db.Begin()
+
+	txn.Discard() // Should not panic
```

Comment:
What do you think about using testify here?

```suggestion
	require.NotPanics(t, txn.Discard)
```